bibcheck/plugins/geocode_all_the_things.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Add latitude and longitude to all records that should have them."""
+
+import os
+
+import googlemaps
+
+from invenio.bibrecord import record_get_field_value, record_get_field_values
+from invenio.config import CFG_GOOGLE_MAPS_API_KEY
+
+
+GMAPS = googlemaps.Client(key=CFG_GOOGLE_MAPS_API_KEY)
+
+
+def geocode(record):
+    """Uses the Google Maps API to geocode a record."""
+    def _get_approximate_address(record):
+        if len(list(record.iterfield('111__c'))):
+            result = record_get_field_value(
+                record, '111', code='c').split(', ')
+        else:
+            city = record_get_field_value(record, '371', code='b')
+            zipcode = record_get_field_value(record, '371', code='e')
+            country = record_get_field_value(record, '371', code='d')
+
+            addresses = [
+                el for el in record_get_field_values(record, '371', code='a')
+                if zipcode not in el]
+            addresses.extend([city, zipcode, country])
+
+            result = [el for el in addresses if el]
+
+        return result
+
+    approximate_address = _get_approximate_address(record)
+    while approximate_address:
+        response = GMAPS.geocode(', '.join(approximate_address))
```

Comment:
After we reach the Google API's limit, what will happen? raised exception? I need to think how to do properly stop the process.